crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -324,6 +324,76 @@ def services
       end
       handle_asynchronously :services
 
+      def openstackbackup
+        dump_location = "/var/lib/crowbar/upgrade/6-to-7-openstack_dump.sql"
+        if File.exist?(dump_location)
+          Rails.logger.debug("OpenStack backup already exists. Skipping...")
+          return
+        end
+
+        db_node = ::Node.find("roles:database-config-default").first
+        user = "postgres"
+        pass = db_node[:postgresql][:password][:postgres]
+        host = db_node[:postgresql].config[:listen_addresses]
+        query = "SELECT SUM(pg_database_size(pg_database.datname)) FROM pg_database;"
+        cmd = "PGPASSWORD=#{pass} psql -t -h #{host} -U #{user} -c '#{query}'"
+
+        Rails.logger.debug("Checking size of OpenStack database")
+        db_size = run_cmd(cmd)
+        unless db_size[:exit_code].zero?
+          Rails.logger.error(
+            "Failed to check size of OpenStack database: #{db_dump[:stdout_and_stderr]}"
+          )
+          raise ::Crowbar::Error::UpgradeDatabaseSizeError.new(
+            db_size[:stdout_and_stderr]
+          )
+        end
+
+        free_space = run_cmd("df -B1 -l --output='avail' / | tail -n1")
```

Comment:
I'm also not sure if the last item is always the one we need to know... in one of my tests this is giving the result for tmpfs